### ISSUE TITLE
SetPartitionsRevokedHandler documentation not clear

### ISSUE THREAD

**Issue by pgrm** [confluentinc/confluent-kafka-dotnet#984]
Description
===========

I'm not trying to be picky, some things in the documentation aren't 100% clear and before assuming I thought to ask:

https://github.com/confluentinc/confluent-kafka-dotnet/blob/026d320aef3959c83045b4daec6c1f2cf4719f81/src/Confluent.Kafka/ConsumerBuilder.cs#L313-L314

This part of the documentation is on the overload of the SetPartitionsRevokedHandler with an `Action` parameter, that should probably go to the overload with the `Func` parameter, I assume?

--------

https://github.com/confluentinc/confluent-kafka-dotnet/blob/026d320aef3959c83045b4daec6c1f2cf4719f81/src/Confluent.Kafka/ConsumerBuilder.cs#L308-L311

This seems contradicting to me given the following example:

https://github.com/confluentinc/confluent-kafka-dotnet/blob/49a217036e4dc3dbf36cc56d470d80c543c7eea4/examples/Consumer/Program.cs#L71-L74

Looking at the above example, I'd assume the second parameter are the partitions which will be revoked, so I should stop any ongoing (async) processing of those partitions. But the documentation states that I get all the partitions.

Maybe the internal implementation of kafka first revokes all the partitions and afterwards assigns them from scratch again? That would explain how those 2 seemingly contradicting examples would actually make sense. If that were to be the case, could you maybe mention it with a side note? Otherwise could you clarify the documentation?

I've tried to find additional info online, but was out of luck to conclusively confirm my assumption. I checked:
- https://kafka.apache.org/11/javadoc/org/apache/kafka/clients/consumer/ConsumerRebalanceListener.html#onPartitionsRevoked-java.util.Collection-
- https://alexandreesl.com/2018/01/21/kafka-implementing-stream-architectures/

Thank you!

**Comment by mhowlett** [comment]
yes, all of your assumptions are correct - thanks - we'll update the documentation.

**Comment by pgrm** [comment]
thx @mhowlett 

As I'm building my first kafka consuming service (ever), I would have one more question, if I understand correctly how the client works.

- my consumer config has `EnableAutoOffsetStore` set to `false`
- I consume the topics and forward them internally into an Rx.Net queue for asynchronous processing (this part can be parallelized per topic / partition, but within each of them, they are processed in order)
- I store the offset manually after the item has been consumed

If I understand it correctly, this might lead to the following situation: Having several unprocessed items in the queue while a partition re-balancing might be triggered at which point I loose all partitions at first, and afterwards gain them again. 

When I start consuming after the partition re-balancing, do I start from the point of my last commit? Or does the client still remember the last consumed offset (even though it's not stored in zookeeper yet) and keeps on consuming from there. In other words - if event `e1` was in the Rx queue while re-balancing happened, and `e1` is in the partition which my consumer gets assigned again after re-balancing, will I see `e1` again, when I call `consumer.Consume()` or only `e2`?

Both cases are OK for me, I just need to know the implementation detail, to understand, how to clear the Queue best. Thanks
